gui/wxpython/lmgr/workspace.py

```diff
@@ -0,0 +1,476 @@
+"""
+@package lmgr::workspace
+
+@brief Workspace manager class for creating, loading and saving workspaces
+
+Class:
+ - lmgr::WorkspaceManager
+
+(C) 2021 by the GRASS Development Team
+
+This program is free software under the GNU General Public
+License (>=v2). Read the file COPYING that comes with GRASS
+for details.
+"""
+
+import sys
+import os
+import tempfile
+
+try:
+    import xml.etree.ElementTree as etree
+except ImportError:
+    import elementtree.ElementTree as etree  # Python <= 2.4
+
+from core import globalvar
+import wx
+import wx.aui
+
+if os.path.join(globalvar.ETCDIR, "python") not in sys.path:
+    sys.path.append(os.path.join(globalvar.ETCDIR, "python"))
+
+from core.settings import UserSettings
+from core.gcmd import RunCommand, GError, GMessage
+from core.workspace import ProcessWorkspaceFile, WriteWorkspaceFile
+from core.debug import Debug
+
+
+class WorkspaceManager:
+    """Workspace Manager for creating, loading and saving workspaces."""
+
+    def __init__(self, lmgr, giface):
+
+        self.lmgr = lmgr
+        self._giface = giface
+        self.workspaceChanged = False  # track changes in workspace
+        self.loadingWorkspace = False
+
+        # load workspace file if requested
+        if self.lmgr.workspaceFile:
+            # load given workspace file
+            if self.Load(self.lmgr.workspaceFile):
+                self.lmgr._setTitle()
+            else:
+                self.lmgr.workspaceFile = None
+        else:
+            # start default initial display
+            self.lmgr.NewDisplay(show=False)
```

Comment:
That's not what I meant, I think it was better before. workspaceFile should be attribute of WorkspaceManager. Loading doesn't need to be in the constructor, you only set workspaceFile to None, then load the file (if requested) from lmgr. Load method should then save the file into workspaceFile. The point is the manager manages the state of workspace file, not lmgr, lmgr just tells it to load a file.

Reply:
Ok @petrasovaa, This makes sense. What do you think now, after the last commit?

Reply:
This looks good!